control-plane/subcommand/server-acl-init/command_test.go

```diff
@@ -2353,6 +2353,82 @@ func TestRun_PoliciesAndBindingRulesACLLoginInSecondaryDatacenterDuringFederatio
 	}
 }
 
+// Test creating the correct ACL policies and Binding Rules for components that use the auth method.
+// The test works by running the command and then ensuring that:
+// * An ACLBindingRule exists which references the ACLRole.
+// * An ACLRole exists and has the correct PolicyName in it's ACLPolicyLinkRule list.
+// * The ACLPolicy exists.
+func TestRun_Login(t *testing.T) {
+	t.Parallel()
+
+	cases := []struct {
+		ComponentName string
+		TokenFlags    []string
+		PolicyNames   []string
+		Roles         []string
+	}{
+		{
+			ComponentName: "controller",
+			TokenFlags:    []string{"-create-controller-token"},
+			PolicyNames:   []string{"controller-policy"},
+			Roles:         []string{resourcePrefix + "-controller-acl-role"},
+		},
+	}
+	for _, c := range cases {
+		t.Run(c.ComponentName, func(t *testing.T) {
+			k8s, testSvr := completeSetup(t)
+			defer testSvr.Stop()
+			_, jwtToken := setUpK8sServiceAccount(t, k8s, ns)
+			k8sMockServer := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+				w.Header().Set("content-type", "application/json")
+				if r != nil && r.URL.Path == "/apis/authentication.k8s.io/v1/tokenreviews" && r.Method == "POST" {
+					w.Write([]byte(tokenReviewsResponse(c.ComponentName, ns)))
+				}
+				if r != nil && r.URL.Path == fmt.Sprintf("/api/v1/namespaces/%s/serviceaccounts/%s", ns, c.ComponentName) &&
+					r.Method == "GET" {
+					w.Write([]byte(serviceAccountGetResponse(c.ComponentName, ns)))
+				}
```

Comment:
These need to be tied to the service account name instead of the `c.ComponentName`. We add a prefix to the service account for the binding rule (i.e. `fmt.Sprintf("%s-%s", resourcePrefix, c.ComponentName)` so probably consul returns 403 because there's no matching binding rule?

Reply:
GENIUS! I got the tests working! ill finish up the PR and have it ready for review on Monday!

Thank you so much @ishustava and @kschoche !! I couldn't have done this without your help.